From 6f97c42ac63c0e46007268acdd8af1e44b31f5d5 Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Tue, 2 Jun 2015 16:26:54 +0800
Subject: [PATCH 415/462] ov680: fix KW issues

Suspicious dereference of pointer 'sd' by passing argument 1 to
function 'v4l2_get_subdevdata'

Change-Id: I4efb259a8866e1887a3ca9fff1781eb97623f73b
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-35074
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/58476
Approver: Bingbu Cao <bingbu.cao@intel.com>
Reviewed-by: Bingbu Cao <bingbu.cao@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Build: IMIN (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 drivers/external_drivers/camera/drivers/media/i2c/ov680.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov680.c b/drivers/external_drivers/camera/drivers/media/i2c/ov680.c
index 8743337..d89260c 100644
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov680.c
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov680.c
@@ -571,10 +571,16 @@ static int __flisclk_ctrl(struct v4l2_subdev *sd, int on)
 
 static int __ov680_s_power(struct v4l2_subdev *sd, int on, int load_fw)
 {
-	struct ov680_device *dev = to_ov680_device(sd);
-	struct i2c_client *client = v4l2_get_subdevdata(sd);
+	struct ov680_device *dev = NULL;
+	struct i2c_client *client = NULL;
 	int ret;
 
+	if (!sd)
+		return -ENODEV;
+
+	dev = to_ov680_device(sd);
+	client = v4l2_get_subdevdata(sd);
+
 	dev_info(&client->dev, "%s - on-%d.\n", __func__, on);
 
 	ret = __flisclk_ctrl(sd, on);
-- 
1.9.1

